include/gfx/timsort.hpp

```diff
@@ -633,6 +634,25 @@ template <typename RandomAccessIterator, typename Compare> class TimSort {
 
 public:
 
+    static void merge(iter_t const lo, iter_t const mid, iter_t const hi, Compare compare) {
+        GFX_TIMSORT_ASSERT(lo <= mid);
+        GFX_TIMSORT_ASSERT(mid <= hi);
+
+        if (lo == mid || mid == hi) {
+            return; // nothing to do
+        }
+
+        TimSort ts;
+        ts.pushRun(lo, mid - lo);
+        ts.pushRun(mid, hi - mid);
+        GFX_TIMSORT_ASSERT(ts.pending_.size() == 2);
+        ts.mergeAt(0, compare);
+        GFX_TIMSORT_ASSERT(ts.pending_.size() == 1);
+
+        GFX_TIMSORT_LOG("size: " << (hi - lo) << " tmp_.size(): " << ts.tmp_.size()
+                                 << " pending_.size(): " << ts.pending_.size());
```

Comment:
I should probably make this log output more useful by printing the two sub-range sizes, and distinguish it from the identical `timsort` output. Or alternatively remove it altogether...